tests/mutation_test.cpp

```diff
@@ -57,56 +60,150 @@ static int get_total_category_strength( const player &p )
+    const mutation_category_id ursine( "URSINE" );
+    const mutation_category_id mouse( "MOUSE" );
+    const mutation_category_id rat( "RAT" );
+
+    npc dummy;
+
+    // With no mutations, no category is the highest
+    CHECK( dummy.get_highest_category().str().empty() );
+    // All categories are at level 0
+    CHECK( dummy.mutation_category_level[lupine] == 0 );
+    CHECK( dummy.mutation_category_level[feline] == 0 );
+    CHECK( dummy.mutation_category_level[raptor] == 0 );
+    CHECK( dummy.mutation_category_level[ursine] == 0 );
+    CHECK( dummy.mutation_category_level[mouse] == 0 );
+    CHECK( dummy.mutation_category_level[rat] == 0 );
+
+    // SMELLY mutation: Common to LUPINE, FELINE, and MOUSE
+    const trait_id smelly( "SMELLY" );
+    REQUIRE( dummy.mutate_towards( smelly ) );
+    // No category should be highest
+    CHECK( dummy.get_highest_category().str().empty() );
+    // All levels should be equal
+    CHECK( dummy.mutation_category_level[lupine] == dummy.mutation_category_level[feline] );
+    CHECK( dummy.mutation_category_level[lupine] == dummy.mutation_category_level[mouse] );
+
+    // UGLY mutation: Common to LUPINE, FELINE, and RAPTOR
+    const trait_id ugly( "UGLY" );
+    REQUIRE( dummy.mutate_towards( ugly ) );
+    // Still no highest, since LUPINE and FELINE should be tied
+    CHECK( dummy.get_highest_category().str().empty() );
+    // LUPINE and FELINE should be equal level, and both stronger than MOUSE or RAPTOR
+    CHECK( dummy.mutation_category_level[lupine] == dummy.mutation_category_level[feline] );
+    CHECK( dummy.mutation_category_level[lupine] > dummy.mutation_category_level[mouse] );
+    CHECK( dummy.mutation_category_level[lupine] > dummy.mutation_category_level[raptor] );
+
+    // GROWL mutation: Common to LUPINE, RAT, and URSINE
+    const trait_id growl( "GROWL" );
+    REQUIRE( dummy.mutate_towards( growl ) );
+    // LUPINE has the most mutations now, and should now be the strongest category
+    CHECK( dummy.get_highest_category().str() == "LUPINE" );
+    // LUPINE category level should be strictly higher than any other
+    CHECK( dummy.mutation_category_level[lupine] > dummy.mutation_category_level[feline] );
+    CHECK( dummy.mutation_category_level[lupine] > dummy.mutation_category_level[mouse] );
+    CHECK( dummy.mutation_category_level[lupine] > dummy.mutation_category_level[raptor] );
+    CHECK( dummy.mutation_category_level[lupine] > dummy.mutation_category_level[ursine] );
+    CHECK( dummy.mutation_category_level[lupine] > dummy.mutation_category_level[rat] );
+}
+
+// If character has all available mutations in a category (pre- or post-threshold), that should be
+// their strongest/highest category. This test verifies that expectation for every category.
+TEST_CASE( "Having all mutations give correct highest category", "[mutations][strongest]" )
+{
+    for( const std::pair<mutation_category_id, mutation_category_trait> cat :
```

Comment:
I suspect clang 10 might want a const for the first one in this pair here.

Reply:
Done. Tidy suggested making `&cat` a reference as well.